crates/e2e/tests/e2e/partially_fillable_pool.rs

```diff
@@ -114,11 +118,14 @@ async fn test(web3: Web3) {
     );
 
     let metadata_updated = || async {
-        onchain.mint_block().await;
         let order = services.get_order(&uid).await.unwrap();
-        !order.metadata.executed_surplus_fee.is_zero()
+        (!order.metadata.executed_surplus_fee.is_zero()
             && order.metadata.executed_buy_amount != Default::default()
-            && order.metadata.executed_sell_amount != Default::default()
+            && order.metadata.executed_sell_amount != Default::default())
+            || {
+                onchain.mint_block().await;
```

Comment:
Was it required to move the block mining into this awkward scope here?
Would be great if that `|| {mine(); false}` construct could be replaced by simply always mining a block on each iteration (comment applies to multiple places).